tests/keyboard_test.go

```diff
@@ -0,0 +1,111 @@
+package tests
+
+import (
+	_ "embed"
+	"runtime"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/xk6-browser/common"
+	"github.com/grafana/xk6-browser/keyboardlayout"
+)
+
+func TestKeyboardPress(t *testing.T) {
+	tb := newTestBrowser(t)
+
+	t.Run("all_keys", func(t *testing.T) {
+		p := tb.NewPage(nil)
+		kb := p.(*common.Page).Keyboard //nolint: forcetypeassert
+		layout := keyboardlayout.GetKeyboardLayout("us")
+
+		assert.NotPanics(t, func() {
+			for k := range layout.Keys {
+				kb.Press(string(k), nil)
+			}
+		})
+	})
+
+	t.Run("backspace", func(t *testing.T) {
+		p := tb.NewPage(nil)
+		kb := p.(*common.Page).Keyboard //nolint: forcetypeassert
+
+		p.SetContent(`<input type="textarea">`, nil)
+		el := p.Query("input")
+		p.Focus("input", nil)
+
+		kb.Type("Hello World!", nil)
+		require.Equal(t, "Hello World!", el.InputValue(nil))
+
+		kb.Press("Backspace", nil)
+		assert.Equal(t, "Hello World", el.InputValue(nil))
+	})
+
+	// Replicates the test from https://playwright.dev/docs/api/class-keyboard
+	t.Run("hold", func(t *testing.T) {
+		t.Skip("FIXME")
+		p := tb.NewPage(nil)
+		kb := p.(*common.Page).Keyboard //nolint: forcetypeassert
+
+		p.SetContent(`<input type="textarea">`, nil)
+		el := p.Query("input")
+		p.Focus("input", nil)
+
+		kb.Type("Hello World!", nil)
+		require.Equal(t, "Hello World!", el.InputValue(nil))
+
+		kb.Press("ArrowLeft", nil)
+		// Should hold the key until Up() is called.
+		kb.Down("Shift")
+		for i := 0; i < len(" World"); i++ {
+			kb.Press("ArrowLeft", nil)
+		}
+		// Should release the key but the selection should remain active.
+		kb.Up("Shift")
+		// Should delete the selection.
+		kb.Press("Backspace", nil)
+
+		assert.Equal(t, "Hello!", el.InputValue(nil))
+	})
+
+	t.Run("combo", func(t *testing.T) {
+		t.Skip("FIXME")
+		p := tb.NewPage(nil)
+		kb := p.(*common.Page).Keyboard //nolint: forcetypeassert
+
+		p.SetContent(`<input type="textarea">`, nil)
+		el := p.Query("input")
+		p.Focus("input", nil)
+
+		kb.Press("Shift+KeyA", nil)
+		kb.Press("Shift+b", nil)
+		kb.Press("C", nil)
+		kb.Press("d", nil)
+		require.Equal(t, "AbCd", el.InputValue(nil))
+
+		metaKey := "Control"
+		if runtime.GOOS == "darwin" {
+			metaKey = "Meta"
+		}
+		kb.Press(metaKey+"+A", nil)
+		kb.Press("Delete", nil)
+		assert.Equal(t, "", el.InputValue(nil))
+	})
+
+	t.Run("newline", func(t *testing.T) {
+		t.Skip("FIXME")
+		p := tb.NewPage(nil)
+		kb := p.(*common.Page).Keyboard //nolint: forcetypeassert
+
+		p.SetContent(`<input type="textarea">`, nil)
+		el := p.Query("input")
+		p.Focus("input", nil)
+
+		kb.Type("Hello", nil)
+		kb.Press("Enter", nil)
+		kb.Press("Enter", nil)
+		kb.Type("World!", nil)
+		assert.Equal(t, "Hello\n\nWorld!", el.InputValue(nil))
+	})
```

Comment:
Strange... This test actually works as expected in JS, but for some reason fails in Go.

This works fine:

```javascript
page.goto('https://mdbootstrap.com/docs/standard/extended/textarea/', { waitUntil: 'networkidle' });
const el = page.$('#textAreaExample1');
el.focus();

page.keyboard.type('Hello');
page.keyboard.press('Enter')
page.keyboard.press('Enter')
page.keyboard.type('World!')

check(el, {
  'value': el.inputValue() == 'Hello\n\nWorld!',
});
```

But this Go test fails with:

```
expected: "Hello\n\nWorld!"
actual  : "HelloWorld!"
```

Any ideas why? :thinking:

Reply:
@imiric actually the `https://mdbootstrap.com/docs/standard/extended/textarea/` uses `textarea` and in our test we use an `input` 

```go
p.SetContent(`<input type="textarea">`, nil)
```

It seems like if we switch to it, the test will pass

Reply:
:facepalm: Thanks, I overlooked that and for some reason thought they would be equivalent.

I'm wondering if there's any value in the test, but I guess it doesn't hurt leaving it.

Reply:
>I'm wondering if there's any value in the test, but I guess it doesn't hurt leaving it.

Well, when I checked this locally, it seems like this is the first and the only usage of `Enter,` so since it's an important key, I'd say that this test is an important one for now at least :sweat_smile:

Reply:
Resolved in latest push (032f45b).